packages/browser/__tests__/Session.spec.ts

```diff
@@ -267,6 +267,158 @@ describe("Session", () => {
+        expect(mySession.info.isLoggedIn).toEqual(false);
+        expect(mySession.info.webId).toBeUndefined();
+        expect(
+          clientAuthentication.handleIncomingRedirect
+        ).toHaveBeenCalledTimes(1);
+      });
+
+      it("re-initialises the user's WebID without redirection if the proper cookie is set", async () => {
+        // Mock localStorage's `getItem` method:
+        const existingLocalStorage = window.localStorage;
+        const localStorageMock = {
+          getItem: jest.fn(() =>
+            JSON.stringify({
+              webId: "https://my.pod/profile#me",
+              sessions: {
+                "https://my.pod/": { expiration: 9000000000000 },
+              },
+            })
+          ),
+        };
+
+        Object.defineProperty(window, "localStorage", {
+          value: localStorageMock,
+          writable: true,
+        });
+        const clientAuthentication = mockClientAuthentication();
+        clientAuthentication.handleIncomingRedirect = jest.fn();
+        const mySession = new Session({ clientAuthentication });
+        await mySession.handleIncomingRedirect("https://some.url");
+        expect(mySession.info.isLoggedIn).toEqual(true);
+        expect(mySession.info.webId).toEqual("https://my.pod/profile#me");
+        expect(
+          clientAuthentication.handleIncomingRedirect
+        ).toHaveBeenCalledTimes(0);
+
+        // Remove the mocked method:
+        Object.defineProperty(window, "localStorage", {
+          value: existingLocalStorage,
+        });
+      });
+
+      it("does not mark an almost-expired session as logged in", async () => {
+        // Mock localStorage's `getItem` method:
+        const existingLocalStorage = window.localStorage;
+        const localStorageMock = {
+          getItem: jest.fn(() =>
+            JSON.stringify({
+              webId: "https://my.pod/profile#me",
+              sessions: {
+                "https://my.pod/": { expiration: Date.now() + 10000 },
+              },
+            })
+          ),
+        };
+
+        Object.defineProperty(window, "localStorage", {
+          value: localStorageMock,
+          writable: true,
+        });
+        const clientAuthentication = mockClientAuthentication();
+        clientAuthentication.handleIncomingRedirect = jest.fn();
+        const mySession = new Session({ clientAuthentication });
+        await mySession.handleIncomingRedirect("https://some.url");
+        expect(mySession.info.isLoggedIn).toEqual(false);
+        expect(mySession.info.webId).toBeUndefined();
+        expect(
+          clientAuthentication.handleIncomingRedirect
+        ).toHaveBeenCalledTimes(1);
+
+        // Remove the mocked method:
+        Object.defineProperty(window, "localStorage", {
+          value: existingLocalStorage,
+        });
+      });
+
+      it("logs you in even if your Identity Provider is not your Resource server", async () => {
```

Comment:
This is probably going to be a pain to remember and implement, but I think we agreed to always try to refer to 'Solid Identity Providers', i.e:
```suggestion
      it("logs you in even if your Solid Identity Provider is not your Resource server", async () => {
```